src/Phinx/Db/Adapter/PdoAdapter.php

```diff
@@ -330,6 +330,38 @@ public function fetchAll($sql)
     /**
      * {@inheritdoc}
      */
+    public function insert(Table $table, $columns, $data)
+    {
+        $this->startCommandTimer();
+
+        $sql = sprintf(
+            "INSERT INTO %s ",
+            $this->quoteTableName($table->getName())
+        );
+
+        $sql .= "(". implode(', ', array_map(array($this, 'quoteColumnName'), $columns)) . ")";
+        $sql .= " VALUES ";
+
+        $rows = array();
+        foreach ($data as $row) {
+            $rows[] = "(" . implode(', ', array_map(function ($value) {
+                if (is_numeric($value)) {
+                    return $value;
+                }
+                return "'{$value}'";
```

Comment:
There really shouldn't be any string mangling going on here - otherwise this is going to fail hard for any data that contains a quote. This should be based on a pdo statement.